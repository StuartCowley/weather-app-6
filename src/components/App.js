import { useState, useEffect } from "react";

import getForecast from "../requests/getForecast";
import LocationDetails from "./LocationDetails";
import SearchForm from "./SearchForm";
import ForecastSummaries from "./ForecastSummaries";
import ForecastDetails from "./ForecastDetails";

import "../styles/App.css";

function App() {
  const [searchText, setSearchText] = useState("");
  const [errorMessage, setErrorMessage] = useState("");
  const [forecasts, setForecasts] = useState([]);
  const [location, setLocation] = useState({ city: "", country: "" });
  const [selectedDate, setSelectedDate] = useState(0);

  const selectedForecast = forecasts.find(
    (forecast) => forecast.date === selectedDate
  );

  const handleCitySearch = () => {
    getForecast(
      searchText,
      setSelectedDate,
      setForecasts,
      setLocation,
      setErrorMessage
    );
  };

  const handleForecastSelect = (event) => {
    setSelectedDate(parseInt(event.target.dataset.date, 10));
  };

  useEffect(() => {
    getForecast(
      searchText,
      setSelectedDate,
      setForecasts,
      setLocation,
      setErrorMessage
    );
  }, []);

  return (
    <main className="weather-app">
      <LocationDetails
        errorMessage={errorMessage}
        city={location.city}
        country={location.country}
      />

      <SearchForm
        searchText={searchText}
        setSearchText={setSearchText}
        handleCitySearch={handleCitySearch}
      />

      {!errorMessage && (
        <>
          <ForecastSummaries
            forecasts={forecasts}
            handleForecastSelect={handleForecastSelect}
          />

          {selectedForecast && <ForecastDetails forecast={selectedForecast} />}
        </>
      )}
    </main>
  );
}

export default App;
