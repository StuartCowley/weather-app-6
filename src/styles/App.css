.weather-app {
  text-align: center;
  margin: 2.5rem;
}

@media (min-width: 561px) {
  .weather-app {
    text-align: left;
  }
}
