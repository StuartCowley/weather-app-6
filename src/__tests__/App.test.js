import { render, screen } from "@testing-library/react";

import App from "../components/App";

import forecast from "../data/forecast.json";

describe("App", () => {
  it("renders the App component correctly", () => {
    render(<App location={forecast.location} />);

    const h1Element = screen.getByText(/Manchester, UK/i);

    expect(h1Element).toBeInTheDocument();
  });
});
