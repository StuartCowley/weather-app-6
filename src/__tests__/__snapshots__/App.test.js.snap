// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the App component correctly 1`] = `
<DocumentFragment>
  <main
    class="weather-app"
  >
    <section
      class="location-details"
    >
      <h1
        class="location-details__heading"
      >
        , 
      </h1>
    </section>
    <section
      class="search-form"
    >
      <input
        class="search-form__input"
        placeholder="Please enter a location"
        type="text"
        value=""
      />
      <button
        class="search-form__button"
        type="submit"
      >
        Search
      </button>
    </section>
    <section
      class="forecast-summaries"
    />
  </main>
</DocumentFragment>
`;
